sonaranalyzer-dotnet/tests/SonarAnalyzer.UnitTest/Rules/PointersShouldBePrivateTest.cs

```diff
@@ -30,10 +30,13 @@ public class PointersShouldBePrivateTest
     {
         [TestMethod]
         [TestCategory("Rule")]
-        public void PointersShouldBePrivate()
-        {
-            Verifier.VerifyAnalyzer(@"TestCases\PointersShouldBePrivate.cs",
-                new PointersShouldBePrivate());
-        }
+        public void PointersShouldBePrivate() => Verifier.VerifyAnalyzer(@"TestCases\PointersShouldBePrivate.cs", new PointersShouldBePrivate());
+
+#if NET5_0 // Function pointers are supported only by .Net 5 runtime
```

Comment:
hmm, I used function pointers in my DeadStore test, w/o compilation directives... and there was no failure https://github.com/SonarSource/sonar-dotnet/pull/3725/files#diff-e80332c6a531ede00e7288259bbce92e353e19b067cc3111b0ccaff17671403fR87

the error would happen at runtime?

Reply:
I've added the check since I've got this error during analysis:
```
[CS8889]: The target runtime doesn't support extensible or runtime-environment default calling conventions. on line 25
```
on line 25: ```public delegate* unmanaged<int, int> f5;```
I've tried to google it a bit but currently there is not a lot of info about it.

Reply:
this snippet works in sharplab.io , see [here](https://sharplab.io/#v2:EYLgtghgzgLgpgJwDQBMQGoA+ABAzAAmwCZ8BhAWACgBvKq/Bwg7ARgDZCAWfAGQHsA5gAoAlPmr4AvvUYBXAHZQIAMzhdCAVhgBLPvIAKfbfPgJRMhrUqMb+FHAA2cARHgAqADzZOAPnwAHGAQWfABefAAyUgA6fgEAbnwAeiT8ADk9AGM+MH8HbQgTC1tA4LDImLjElPSsnLyCoutGaUpWoA==)
```
public class C
{

    public static void Log() { }
    unsafe void FunctionPointer()
    {
        delegate*<void> ptr1 = &C.Log; // Noncompliant
        ptr1 = &C.Log; // Noncompliant
    }
}
```

Reply:
yes, the problem appears when using `unmanaged` (reproducible also on sharplab.io). I think we prefer to keep the case though to document the behavior when these are used.